lib_expense/src/test/java/com/example/expenselogger/lib_expense/domain/usecase/GetExpenseTest.kt

```diff
@@ -1,38 +1,23 @@
 package com.example.expenselogger.lib_expense.domain.usecase
 
 import com.example.expenselogger.lib_expense.domain.contract.ExpenseContract
-import com.example.expenselogger.lib_expense.domain.fakes.FakeExpenseContract
 import com.example.expenselogger.lib_expense.domain.model.DummyData
 import com.example.expenselogger_test_utils.TestAsyncExecutor
 import com.google.common.truth.Truth.assertThat
 import kotlinx.coroutines.test.runBlockingTest
 import org.junit.Test
+import org.mockito.Mockito.`when`
 import org.mockito.Mockito.mock
-import org.mockito.Mockito.verify
 
 public class GetExpenseTest {
 
-    private val expenseContract = FakeExpenseContract()
-    private val getExpense = GetExpense(
-        expenseContract,
-        TestAsyncExecutor()
-    )
+    private val expenseContract = mock(ExpenseContract::class.java)
+    private val getExpense = GetExpense(expenseContract, TestAsyncExecutor())
```

Comment:
move the creation of the system under test (SUT) and it's dependency into the individual tests. Sharing one instance of the objects across several tests can lead to bugs especially if the tests mutate the state of the `SUT`.
That's not the case here cos the SUT is stateless. Nonetheless it is good practice to either initialise the objects inside the `@before` annotated function block, or to do it within the tests themselves

Reply:
Okay, thank you! I will correct it